testprojects/src/python/unicode/compilation_failure/BUILD

```diff
@@ -1,8 +1,14 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-
 python_binary(
   sources = ['main.py'],
-  tags={"nolint"},
+  dependencies = [':lib'],
+  tags = {"nolint"},
+)
+
+python_library(
+  name='lib',
+  sources = ['__init__.py'],
+  tags = {"nolint"},
 )
```

Comment:
Ohhh I see the bug now! So, the `__init__` inference rule found that there is indeed an owner of this file, but that owner isn't an actual Python target. It would be more robust if we instead had the inference rule filter out all the resulting owners to check that `.has_field(PythonSources)` is true.

Without this target definition added, we would have wanted the `__init__.py` inference rule to error that there is no _Python_ owner of this file.

--

It's fine to fix this in a followup, imo, so long as we do fix it.

Reply:
https://github.com/pantsbuild/pants/issues/10467